src/mcu.c

```diff
@@ -161,6 +161,11 @@ void timer_task() {
         timer_4hz = ((timer_cnt & 3) == 3);
         timer_8hz = ((timer_cnt & 1) == 1);
         timer_16hz = 1;
+    } else {
```

Comment:
Just a comment:
I think this timer handling should be cleaned to use only `timer_cnt` value and then use macros to check when certain time has elapsed to make sure it works every time. This implementation worries me a bit at the moment... Another PR though.